packages/create-redwood-app/templates/ts/scripts/seed.ts

```diff
@@ -1,67 +1,29 @@
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
 import type { Prisma } from '@prisma/client'
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
 import { db } from 'api/src/lib/db'
 
+// Manually apply seeds via the `yarn rw prisma db seed` command.
+//
+// Seeds automatically run the first time you run the `yarn rw prisma migrate dev`
+// command and every time you run the `yarn rw prisma migrate reset` command.
+//
+// See https://redwoodjs.com/docs/database-seeds for more info
+
 export default async () => {
   try {
-    //
-    // Manually seed via `yarn rw prisma db seed`
-    // Seeds automatically with `yarn rw prisma migrate dev` and `yarn rw prisma migrate reset`
-    //
-    // Update "const data = []" to match your data model and seeding needs
-    //
-    const data: Prisma.UserExampleCreateArgs['data'][] = [
-      // To try this example data with the UserExample model in schema.prisma,
-      // uncomment the lines below and run 'yarn rw prisma migrate dev'
-      //
-      // { name: 'alice', email: 'alice@example.com' },
-      // { name: 'mark', email: 'mark@example.com' },
-      // { name: 'jackie', email: 'jackie@example.com' },
-      // { name: 'bob', email: 'bob@example.com' },
-    ]
-    console.log(
-      "\nUsing the default './scripts/seed.ts' template\nEdit the file to add seed data\n"
-    )
+    const data = []
 
-    if ((await db.userExample.count()) === 0) {
-      // Note: if using PostgreSQL, using `createMany` to insert multiple records is much faster
-      // @see: https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#createmany
-      await Promise.all(
-        //
-        // Change to match your data model and seeding needs
-        //
-        data.map(async (data: Prisma.UserExampleCreateArgs['data']) => {
-          const record = await db.userExample.create({ data })
-          console.log(record)
-        })
-      )
-    } else {
-      console.log('Users already seeded')
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    for (const item of data) {
+      // Create a database record for each item in `data`
+      // ie: `await db.user.create({ data: item })`
```

Comment:
So in the docs I talk about the seeds being idempotent, and running the file multiple times won't create new records. Unfortunately with `createMany()` it'll do just that. :( In the doc I talk about the tradeoffs and a fallback so that you *can* use `createMany()` but that you'd want to check first to make sure you're not creating duplicate records.

Reply:
Although that makes me realize even THIS code example should use `upsert()` instead of `create()` so follow that recommendation!

Reply:
Oh, yes. I would still check. 

I just meant not looping and await each create.


```
if ((await db.user.count()) === 0) {
      await db.user.createMany({ data: users })
    } else {
      console.log('Users already seeded')
    }
```

```    if ((await db.post.count()) === 0) {
      await db.post.createMany({ data: posts })
    } else {
      console.log('Posts already seeded')
    }```

Reply:
I'd leave a check for multiple `seed` runs and the `createMany` for initial db seed and db `migration resets`.

Reply:
Ahh, I was going even more granular than that and checking that each individual record was present in the database before creating/updating. All the time when I'm in dev and things are getting updated and deleted all over the place, I can always re-run the seed and be back to a known state with those fixed seeds: I don't want to completely recreate the database from scratch and start over, just get me back to a known minimum amount of data.

> I'd leave a check for multiple seed runs and the createMany for initial db seed and db migration resets

How would you know from within the seed script that it's the initial/reset run and not a manual/multiple run?

Reply:
Okay you convinced me 😃 How's this?

```
// Create your database records here! For example, seed some users:
//
// const users = [
//   { name: 'Alice', email: 'alice@redwoodjs.com },
//   { name: 'Bob', email: 'bob@redwoodjs.com },
// ]
//
// await db.user.createMany({ data: users })
```